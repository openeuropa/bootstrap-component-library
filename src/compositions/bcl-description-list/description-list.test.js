import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-description-list/data/data";
import demoDataHorizontal from "@openeuropa/bcl-description-list/data/data--horizontal";
const template = "@oe-bcl/bcl-description-list/description-list.html.twig";

const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Description List", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);
    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test(`horizontal renders correctly`, () => {
    expect.assertions(1);
    return expect(render(demoDataHorizontal)).resolves.toMatchSnapshot();
  });
});

describe("OE - Description List with title", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);
    return expect(
      render({
        ...demoData,
        title: "Description list test title",
        title_tag: "h6",
      })
    ).resolves.toMatchSnapshot();
  });

  test(`horizontal renders correctly`, () => {
    expect.assertions(1);
    return expect(
      render({
        ...demoDataHorizontal,
        title: "Description list test title",
        title_tag: "h6",
      })
    ).resolves.toMatchSnapshot();
  });
});
