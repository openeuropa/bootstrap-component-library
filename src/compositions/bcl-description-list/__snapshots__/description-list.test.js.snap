// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Description List horizontal renders correctly 1`] = `
<jest>
  <dl
    class="d-md-grid grid-3-9"
  >
    <dt
      class="d-flex"
    >
      <svg
        class="me-2 mt-1 bi icon--xs"
      >
        <use
          xlink:href="/icons.svg#geo-alt-fill"
        />
      </svg>
      <p
        class="mb-0 d-inline-block"
      >
        Only title label with icon
      </p>
    </dt>
    <dd>
      Description text goes here.
    </dd>
    <div>
      <dt>
        <p
          class="mb-0"
        >
          First title label
        </p>
      </dt>
      <dt
        class="d-flex"
      >
        <svg
          class="me-2 mt-1 bi icon--xs"
        >
          <use
            xlink:href="/icons.svg#geo-alt-fill"
          />
        </svg>
        <p
          class="mb-0 d-inline-block"
        >
          Second title label
        </p>
      </dt>
    </div>
    <dd>
      <p
        class="mb-0"
      >
        Description text goes here.
      </p>
    </dd>
    <dt>
      
          Only title label
    </dt>
    <div>
      <dd>
        <p
          class="mb-2"
        >
          Description text goes here.
        </p>
      </dd>
      <dd>
        <p
          class="mb-2"
        >
          Description text goes here.
        </p>
      </dd>
    </div>
  </dl>
</jest>
`;

exports[`OE - Description List renders correctly 1`] = `
<jest>
  <dl>
    <dt>
      <svg
        class="me-2 bi icon--xs"
      >
        <use
          xlink:href="/icons.svg#geo-alt-fill"
        />
      </svg>
      <p
        class="d-inline-block mb-0"
      >
        Only title label with icon
      </p>
    </dt>
    <dd>
      <p
        class="mb-2"
      >
        Description text goes here.
      </p>
    </dd>
    <dt>
      First title label
    </dt>
    <dt>
      <svg
        class="me-2 bi icon--xs"
      >
        <use
          xlink:href="/icons.svg#geo-alt-fill"
        />
      </svg>
      <p
        class="mb-0 d-inline-block"
      >
        Second title label
      </p>
    </dt>
    <dd>
      <p
        class="mb-2"
      >
        Description text goes here.
      </p>
    </dd>
    <dt>
      
          Only title label
    </dt>
    <dd>
      <p
        class="mb-2"
      >
        Description text goes here.
      </p>
    </dd>
    <dd>
      <p
        class="mb-2"
      >
        Description text goes here.
      </p>
    </dd>
  </dl>
</jest>
`;

exports[`OE - Description List with title horizontal renders correctly 1`] = `
<jest>
  <h6
    class="fw-bold mb-4"
  >
    Description list test title
  </h6>
  <dl
    class="d-md-grid grid-3-9"
  >
    <dt
      class="d-flex"
    >
      <svg
        class="me-2 mt-1 bi icon--xs"
      >
        <use
          xlink:href="/icons.svg#geo-alt-fill"
        />
      </svg>
      <p
        class="mb-0 d-inline-block"
      >
        Only title label with icon
      </p>
    </dt>
    <dd>
      Description text goes here.
    </dd>
    <div>
      <dt>
        <p
          class="mb-0"
        >
          First title label
        </p>
      </dt>
      <dt
        class="d-flex"
      >
        <svg
          class="me-2 mt-1 bi icon--xs"
        >
          <use
            xlink:href="/icons.svg#geo-alt-fill"
          />
        </svg>
        <p
          class="mb-0 d-inline-block"
        >
          Second title label
        </p>
      </dt>
    </div>
    <dd>
      <p
        class="mb-0"
      >
        Description text goes here.
      </p>
    </dd>
    <dt>
      
          Only title label
    </dt>
    <div>
      <dd>
        <p
          class="mb-2"
        >
          Description text goes here.
        </p>
      </dd>
      <dd>
        <p
          class="mb-2"
        >
          Description text goes here.
        </p>
      </dd>
    </div>
  </dl>
</jest>
`;

exports[`OE - Description List with title renders correctly 1`] = `
<jest>
  <h6
    class="fw-bold mb-4"
  >
    Description list test title
  </h6>
  <dl>
    <dt>
      <svg
        class="me-2 bi icon--xs"
      >
        <use
          xlink:href="/icons.svg#geo-alt-fill"
        />
      </svg>
      <p
        class="d-inline-block mb-0"
      >
        Only title label with icon
      </p>
    </dt>
    <dd>
      <p
        class="mb-2"
      >
        Description text goes here.
      </p>
    </dd>
    <dt>
      First title label
    </dt>
    <dt>
      <svg
        class="me-2 bi icon--xs"
      >
        <use
          xlink:href="/icons.svg#geo-alt-fill"
        />
      </svg>
      <p
        class="mb-0 d-inline-block"
      >
        Second title label
      </p>
    </dt>
    <dd>
      <p
        class="mb-2"
      >
        Description text goes here.
      </p>
    </dd>
    <dt>
      
          Only title label
    </dt>
    <dd>
      <p
        class="mb-2"
      >
        Description text goes here.
      </p>
    </dd>
    <dd>
      <p
        class="mb-2"
      >
        Description text goes here.
      </p>
    </dd>
  </dl>
</jest>
`;
